src/Spanner/Result.php

```diff
@@ -0,0 +1,326 @@
+     * this result set.
+     *
+     * Stats are not returned by default and will not be accessible until the
+     * entire set of results has been iterated through.
+     *
+     * Example:
+     * ```
+     * $stats = $result->stats();
+     * ```
+     *
+     * ```
+     * // Executing a query with stats returned.
+     * $res = $database->execute('SELECT * FROM Posts', [
+     *     'queryMode' => 'PROFILE'
+     * ]);
+     * ```
+     *
+     * @codingStandardsIgnoreStart
+     * @return array|null [ResultSetStats](https://cloud.google.com/spanner/docs/reference/rpc/google.spanner.v1#google.spanner.v1.ResultSetStats).
+     * @codingStandardsIgnoreEnd
+     */
+    public function stats()
+    {
+        return $this->stats;
+    }
+
+    /**
+     * Returns a snapshot which was begun in the read or execute, if one exists.
+     *
+     * Will be populated once the result set is iterated upon.
+     *
+     * Example:
+     * ```
+     * $snapshot = $result->snapshot();
+     * ```
+     *
+     * @return Snapshot|null
+     */
+    public function snapshot()
+    {
+        return $this->snapshot;
+    }
+
+    /**
+     * Returns a transaction which was begun in the read or execute, if one exists.
+     *
+     * Will be populated once the result set is iterated upon.
+     *
+     * Example:
+     * ```
+     * $transaction = $result->transaction();
+     * ```
+     *
+     * @return Transaction|null
+     */
+    public function transaction()
+    {
+        return $this->transaction;
+    }
+
+    /**
+     * @access private
+     */
+    public function getIterator()
+    {
+        return $this->rows();
+    }
+
+    /**
+     * Yields rows from a partial result set.
+     *
+     * @return \Generator
+     */
+    private function getRowsFromPartial(array $partial)
+    {
+        $this->stats = isset($partial['stats']) ? $partial['stats'] : null;
+        $this->resumeToken = isset($partial['resumeToken']) ? $partial['resumeToken'] : null;
+
+        if (isset($partial['metadata'])) {
+            $this->metadata = $partial['metadata'];
+            $this->columns = $partial['metadata']['rowType']['fields'];
+        }
+
+        if (isset($partial['metadata']['transaction']['id'])) {
+            if ($this->transactionContext === SessionPoolInterface::CONTEXT_READ) {
+                $this->snapshot = $this->operation->createSnapshot(
+                    $this->session,
+                    $partial['metadata']['transaction']
+                );
+            } else {
+                $this->transaction = $this->operation->createTransaction(
+                    $this->session,
+                    $partial['metadata']['transaction']
+                );
+            }
+        }
+
+        if ($this->cachedValues) {
+            $partial['values'] = $this->mergeValues($this->cachedValues, $partial['values']);
+            $this->cachedValues = null;
+        }
+
+        if (isset($partial['chunkedValue'])) {
```

Comment:
PartialResultSet might still contain one or more complete rows. Chunked just means that the last value in the result is chunked. So you should still yield any complete rows in the PartialResultSet. 
In worse case if every single PartialResultSet has chunked set to true except the last one, you will just keep on caching everything till you get the last chunk. That is not ideal.